### PR TITLE
Merge of feat/messages to master

### PR DESCRIPTION
**Pull request by nrebhun**
## Summary:
- handledata.js
    - convert _messageThread_ (array of MessageThread objects) in Player class to _conversation_ (key-value object)
    - simplify _sendMessage_ function

### CODE REVIEW

File: handleData.js [hunk 1]
```diff
@@ -91,6 +91,20 @@ Player.prototype.printAddressBook = function() {
...
+        this.conversation[recipient.playerName] = [content];
+    }
+
+    if (this.playerName in recipient.conversation) {
+        recipient.conversation[this.playerName].push(content);
+    } else {
+        recipient.conversation[this.playerName] = [content];
+    }
+
+    /* Old method for player lookup and conversation establishment:
```
Comment (zmbush) [inline handleData.js]: This is unnecessary to keep. Git history lets you look into the past so you can recover the code later :)

Comment (nrebhun) [inline handleData.js]: Good point.

File: handleData.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ Message.prototype.getFormattedTimestamp = function() {
             this.timeStamp.getSeconds();
 }
 
-// Begin MessageThread Definition
+// Begin MessageThread Definition, MAY NOT BE NECESSARY
 var MessageThread = function(sender, receiver) {
```
Comment (zmbush) [inline handleData.js]: Like I said on chat, it might be nice to rename this to Conversation.

**Comment by zmbush** [comment]
LGTM

**Pull request merged**